### PR TITLE
refactor(copy): improved copy

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
- improved thread-safe and clean up code on copy/move/rename

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- zboxcli: https://github.com/0chain/zboxcli/pull/303

### CODE REVIEW

File: zboxcore/sdk/copyworker.go [hunk 1]
```diff
@@ -72,40 +76,67 @@ func (req *CopyRequest) copyBlobberObject(blobber *blockchain.StorageNode, blobb
...
+
 			l.Logger.Info(blobber.Baseurl, " "+req.remotefilepath, " copied.")
-		} else {
-			resp_body, err := ioutil.ReadAll(resp.Body)
-			if err == nil {
-				l.Logger.Error(blobber.Baseurl, "Response: ", string(resp_body))
+			return nil
+		}
+
+		resp_body, err := ioutil.ReadAll(resp.Body)
```
Comment (peterlimg) [inline zboxcore/sdk/copyworker.go:79]: Use respBody

File: zboxcore/sdk/renameworker.go [hunk 2]
```diff
@@ -71,17 +74,22 @@ func (req *RenameRequest) renameBlobberObject(blobber *blockchain.StorageNode, b
...
-			req.maskMU.Unlock()
+
 			l.Logger.Info(blobber.Baseurl, " "+req.remotefilepath, " renamed.")
-		} else {
-			resp_body, err := ioutil.ReadAll(resp.Body)
-			if err == nil {
-				l.Logger.Error(blobber.Baseurl, "Response: ", string(resp_body))
+			return nil
+		}
+		resp_body, err := ioutil.ReadAll(resp.Body)
```
Comment (peterlimg) [inline zboxcore/sdk/renameworker.go]: respBody

File: zboxcore/sdk/renameworker.go [hunk 3]
```diff
@@ -90,25 +98,31 @@ func (req *RenameRequest) renameBlobberObject(blobber *blockchain.StorageNode, b
...
+			refEntity, err := req.renameBlobberObject(req.blobbers[blobberIdx])
+
+			if err != nil {
+				l.Logger.Error(err.Error())
+			}
+
+			wait <- RenameResult{
+				BlobberIndex: blobberIdx,
+				FileRef:      refEntity,
+				Renamed:      err == nil,
```
Comment (peterlimg) [inline zboxcore/sdk/renameworker.go]: Seems the RenameResult and CopyResult could be refactored to ProcessResult that with the `Succeed bool` field to indicate whether the err is nil, or just have the `Err error` field that let the caller to check if success. Then we don't need to define result struct respectively for each process type.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: zboxcore/sdk/copyworker.go (hunk 1 above)
Comment (cnlangzi) [reply zboxcore/sdk/copyworker.go:79]: fixed

File: zboxcore/sdk/renameworker.go (hunk 2 above)
Comment (cnlangzi) [reply zboxcore/sdk/renameworker.go]: fixed

File: zboxcore/sdk/renameworker.go (hunk 3 above)
Comment (cnlangzi) [reply zboxcore/sdk/renameworker.go]: fixed

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/improve_copy_perf  |
| zboxcli  | fix/improve_copy_perf  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**